gui/default/syncthing/device/editDeviceModalView.html

```diff
@@ -172,7 +165,7 @@
     <button type="button" class="btn btn-default btn-sm" data-dismiss="modal">
       <span class="fas fa-times"></span>&nbsp;<span translate>Close</span>
     </button>
-    <div ng-if="has(['existing', 'defaults'], currentDevice._editing)" class="pull-left">
+    <div ng-if="editingDeviceExisting()" class="pull-left">
```

Comment:
I just want to add a quick comment here. This is explained in the commit message too, as a part of the removing `currentDevice._editing` from the HTML code, I've also cleaned up the code here. The reason for removing "defaults" is and keeping only "existing" is that right now, there is a "non-functional" remove button when editing device defaults. Clicking it does nothing apart from closing the modal. I don't think this is intentional, and as I see it, the only case where the button makes sense is when editing existing devices.

Reply:
Regarding "stuff explained in the commit message": I'd love to have more of that and I usually write very detailed messages myself.  But in this project it's kind of usual practice to just jot down "wip" or something because it will get squashed and discarded anyway.  That trains people to not look at the individual commits at all I suppose.  Unlike e.g. Linux development, where a PR usually concerns a "patch series".

Reply:
Yeah, although I always copy and paste the content of the main commit message into the PR description too. I also try to keep the message itself as descriptive and detailed as possible. One problem with this approach is that in order to actually edit the commit message I've got to rebase the whole commit history, and then do a force-push instead of just adding newer commit on top of the existing ones.

Reply:
The commit message on your branch does not really matter.  PRs are squash-merged here, so the message for the actual commit in official Syncthing history is written by whoever merges it.  GitHub concatenates all individual commit messages only as a draft to start from (e.g. remove all "wip" lines).

Rebasing helps to keep that draft clean, but in general it's usually more confusing during review.  I often checkout PRs locally for testing, which works with added commits (fast-forward), but not after rebasing.  So best avoided unless you have a good reason (e.g. several logical changes within one PR that each deserve a detailed commit message).

Reply:
From my observation, it depends who merges the commit 😉. The "wips" are always removed, but the commit message itself is often left with no changes. For example, see https://github.com/syncthing/syncthing/commit/0fb7cc186c0ba7262a6cd1404d6aa1c812b7f529 and https://github.com/syncthing/syncthing/commit/abdf02451705a4b606f2991bec2ea631f34229fe. In the first case, both the commit message and the PR description matched, and the commit message was left in the final squash and merge. In the second case though, I updated only the PR description and forgot to update the commit message, which was then still used in the final squash and merge commit, despite not 100% matching the final iteration of the PR.

This is one of the reasons why I prefer to keep both the main commit message and the PR description have the exact same text, and also update both when needed (which is only possible through a `rebase` and `force-push`).

Reply:
If it was my project, I'd have each PR submitter push a cleaned-up history, which then gets merged without squashing and always with a merge commit.  More like the "patch series" discipline on the Linux kernel for example.  But that's not this team's policy and not a discussion to be had here :wink: